### ISSUE TITLE
Remote SSH restarts repeatedly when opening remote git folder

### ISSUE THREAD

**Issue by celynw** [microsoft/vscode-pull-request-github#5782]
Type: <b>Bug</b>

I am opening a folder on a remote computer (Ubuntu 22.04) using the Remote SSH extension. The folder I am opening is a git repository which has checked out a PR which is in the process of being reviewed. The already open tab is a python file with active comments.

After asking for my SSH passphrase, it loads and I can see my files and it opens the same files as I'm used to. The extensions start loading, I can see my git status (line colours, Source Control sidebar). My Python and Ruff extensions load.

Then this PR&I extension loads, and I can even see the inline comments in the open file. I can see on the status bar, it counts down "Files to analyze" which is from the Python extension, I believe (Pylance). When it gets to 1 (maybe after 1, and maybe this is unrelated) a few seconds later, the window reloads. At this point, if I don't have my SSH agent running, it will ask for my SSH passphrase again. After 3 attempts, it gives up on loading extensions, but I do still seem to have an SSH connection and can edit files.

Looking at the Output - Server:

```
#
# Fatal error in , line 0
# Fatal JavaScript invalid size error 169220804
#
#
#
#FailureMessage Object: 0x7ffedbcbfb00

2024-02-26 12:17:29.675 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  1: 0xbf1ac1  [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.675 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  2: 0x1e8cd84 V8_Fatal(char const*, ...) [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.677 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  3: 0xf17a28  [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.678 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  4: 0x10c75e2  [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.678 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  5: 0x10c78b6  [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.679 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  6: 0x12d884b v8::internal::Runtime_GrowArrayElements(int, unsigned long*, v8::internal::Isolate*) [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.680 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910><stderr>  7: 0x1724e39  [/home/user/.vscode-server/bin/903b1e9d8990623e3d7da1df3d33db3e42d80eda/node]
2024-02-26 12:17:29.924 [info] [127.0.0.1][ea515c06][ExtensionHostConnection] <651910> Extension Host Process exited with code: null, signal: SIGTRAP.
2024-02-26 12:17:30.013 [error] [127.0.0.1][ea515c06][ExtensionHostConnection] Unknown reconnection token (seen before).
2024-02-26 12:17:30.282 [info] [127.0.0.1][c7a5ad2b][ExtensionHostConnection] New connection established.
2024-02-26 12:17:30.294 [info] [127.0.0.1][c7a5ad2b][ExtensionHostConnection] <652525> Launched Extension Host Process.
2024-02-26 12:17:45.017 [info] [127.0.0.1][c7a5ad2b][ExtensionHostConnection] <652525><stderr> 
```

I found this issue using the "Extension Bisect" feature of VSCode (awesome!). The problem is totally reproducible by me, and I have had this problem for at least a few days. I have had the extension for a good few months.

Extension version: 0.80.1
VS Code version: Code 1.86.2 (903b1e9d8990623e3d7da1df3d33db3e42d80eda, 2024-02-13T19:40:56.878Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Connection to 'ssh-remote+pppc' could not be established

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600X 6-Core Processor              (12 x 3693)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.91GB (13.89GB free)|
|Process Argv|C:\\Users\\Celyn\\Desktop\\backup.json|
|Screen Reader|no|
|VM|0%|

Connection to 'ssh-remote+pppc' could not be established
</details>

**Comment by alexr00** [comment]
>  a few seconds later, the window reloads. 

This is strange. This extension never reloads the window. Maybe GitHub Pull Requests is doing something that is causing a different extension to do the reload? Could you try the following:
- Disable all extensions except GitHub Pull Requests and Remote SSH
- Try to repro

Does the issue still happen?

**Comment by celynw** [comment]
I can no longer reproduce this, even rolling back to version `0.80.1`. So I'll close this. Thanks for following up!

I was experiencing this problem for at least a few days in different repositories so much I had to disable the extension.
When using VSCode Extension bisect, it was 100% consistent with this extension enabled, even down to no other extensions enabled. I have a feeling the reloading must have been caused by the SSH extension failing and retrying for whatever reason.
